internal/pkg/scorecard/helpers.go

```diff
@@ -35,3 +40,92 @@ func ResultsCumulative(results []TestResult) (earned, max int) {
 	}
 	return earned, max
 }
+
+// CalculateStates updates the state fields of the JSONOut based on the TestResults in out.Tests
+func CalculateStates(out *scapiv1alpha1.ScorecardResult) {
```

Comment:
Let's try not to rely on pointers and mutable state to do things IMO. 

change to `CalculateResult([]...Tests) scapiv1alpha1.ScorecardResult`